src/extensions/lens-main-extension.ts

```diff
@@ -30,15 +29,8 @@ import type { MenuRegistration } from "./registries";
 export class LensMainExtension extends LensExtension {
   appMenus: MenuRegistration[] = [];
 
-  async navigate<P extends object>(pageId?: string, params?: P, frameId?: number) {
-    const windowManager = WindowManager.getInstance();
-    const pageUrl = getExtensionPageUrl({
-      extensionId: this.name,
-      pageId,
-      params: params ?? {}, // compile to url with params
-    });
-
-    await windowManager.navigate(pageUrl, frameId);
+  async navigate(pageId?: string, params?: Record<string, any>, frameId?: number) {
+    return WindowManager.getInstance().navigateExtension(this.id, pageId, params, frameId);
```

Comment:
`getExtensionPageUrl` really shouldn't have ever been accessible from `main` since neither `globalPageMenuRegistry` nor `pageMenuRegistry` would have been populated on `main` ever. 

So while this bug has only been recently noticed (because during #2779 I consciously didn't create instances on `main` for registries that aren't proper to `main`) this functionality has always been broken.